packages/block-library/src/group/inspector.js

```diff
@@ -0,0 +1,232 @@
+
+import {
+	Fragment,
+} from '@wordpress/element';
+
+export default function Inspector( props ) {
+	const {
+		clientId,
+		setBackgroundColor,
+		backgroundColor,
+		attributes,
+		setAttributes,
+	} = props;
+
+	/**
+	 * Resets a single spacing attribute for a given dimension
+	 * (and optionally a given device)
+	 * @param  {string} dimension the dimension property (eg: `padding`)
+	 * @param  {string} device    the device which this dimension applies to (eg: `mobile`, `tablet`)
+	 * @return {void}
+	 */
+	const resetSpacingDimension = ( dimension, device = '' ) => {
+		setAttributes( {
+			[ `${ dimension }${ device }` ]: '',
+		} );
+	};
+
+	/**
+	 * Resets all the responsive attributes for a given dimension
+	 * @param  {string} dimension the dimension property (eg: `padding`)
+	 * @return {void}
+	 */
+	const resetResponsiveSpacingForDimension = ( dimension ) => {
+		dimension = dimension.toLowerCase();
+
+		setAttributes( {
+			[ `${ dimension }SizeMobile` ]: '',
+			[ `${ dimension }SizeTablet` ]: '',
+		} );
+	};
+
+	/**
+	 * Updates the spacing attribute for a given dimension
+	 * (and optionally a given device)
+	 * @param  {string} size      a slug representing a dimension size (eg: `medium`)
+	 * @param  {string} dimension the dimension property (eg: `padding`)
+	 * @param  {string} device    the device which this dimension applies to (eg: `mobile`, `tablet`)
+	 * @return {void}
+	 */
+	const updateSpacing = ( size, dimension, device = '' ) => {
+		setAttributes( {
+			[ `${ dimension }${ device }` ]: size,
+		} );
+	};
+
+	/**
+	 * Toggles the responsive spacing UI for a given dimension
+	 * and clears any responsive attribute
+	 * @param  {string} dimension the dimension property (eg: `padding`)
+	 * @return {void}
+	 */
+	const onToggleResponsiveSpacing = ( dimension ) => {
+		dimension = startCase( dimension );
+
+		const attr = `responsive${ dimension }`;
+		const responsiveDimensionState = ! attributes[ attr ];
+
+		if ( ! responsiveDimensionState ) {
+			resetResponsiveSpacingForDimension( dimension );
+		}
+		setAttributes( {
+			[ attr ]: responsiveDimensionState,
+		} );
+	};
+
+	return (
+		<InspectorControls>
+			<PanelColorSettings
+				title={ __( 'Color Settings' ) }
+				colorSettings={ [
+					{
+						value: backgroundColor.color,
+						onChange: setBackgroundColor,
+						label: __( 'Background Color' ),
+					},
+				] }
+			/>
+
+			<PanelBody title={ __( 'Spacing' ) }>
+
+				<fieldset className="block-editor-responsive-controls">
+					<legend className="block-editor-responsive-controls__label">{ __( 'Padding' ) } </legend>
+					<ToggleControl
+						label={ __( 'Manually adjust Padding based on screensize?' ) }
+						checked={ attributes.responsivePadding }
+						onChange={ () => onToggleResponsiveSpacing( 'padding' ) }
+					/>
+
+					{ ! attributes.responsivePadding && (
+						<DimensionControl
+							title={ __( 'Padding' ) }
+							property="padding"
+							id={ clientId }
```

Comment:
You can ease the requirement to constantly pass a unique `id` into the component by using the `withInstanceId` HOC which other controls are already using:

https://github.com/WordPress/gutenberg/blob/9fcccb3c374238a161fc92dc414f0e11d135606e/packages/components/src/text-control/index.js#L29

Reply:
Great advice. I'd forgotten about that one.

Annoyingly adding this has broken all my tests because even with Enzyme `mount` doesn't render the `<button>` elements now they are wrapped in a HOC.

Reply:
Fixed.